ldap/servers/plugins/acl/acl.c

```diff
@@ -2408,10 +2466,53 @@ acl__resource_match_aci(Acl_PBlock *aclpb, aci_t *aci, int skip_attrEval, int *a
                                                     ACL_EVAL_TARGET_FILTER);
             slapi_ch_free((void **)&lasinfo);
         } else {
-            if (slapi_vattr_filter_test(NULL, aclpb->aclpb_curr_entry,
-                                        aci->targetFilter,
-                                        0 /*don't do access check*/) != 0) {
-                filter_matched = ACL_FALSE;
+            Slapi_DN *sdn;
+            char* attr_evaluated = "None";
+            char logbuf[1024];
+            char *filterstr; /* key to retrieve/add targetfilter value in the cache */
+            struct targetfilter_cached_result *previous_filter_test;
+            
+            /* only usefull for debug purpose */
+            if (aclpb->aclpb_curr_attrEval && aclpb->aclpb_curr_attrEval->attrEval_name) {
+                attr_evaluated = aclpb->aclpb_curr_attrEval->attrEval_name;
+            }
+            sdn = slapi_entry_get_sdn(aclpb->aclpb_curr_entry);
+            
+            /* The key for the cache is the original filter */
+            filterstr = slapi_filter_to_string(aci->targetFilter, logbuf, sizeof (logbuf));
+            
+            previous_filter_test = targetfilter_cache_lookup(aclpb, filterstr);
+            if (previous_filter_test) {
```

Comment:
Is there some evidence of cache hit rates here during this kind of eval? It seems pretty situational maybe?

Reply:
For a given operation, if a selected ACI (with targetfilter) targets N requested attribute then the rate is N/N+1. so its minimal rate is 50% (one attribute requested).

Reply:
Even minumum of 50% is good IMO.